src/resolver.rs

```diff
@@ -292,22 +292,68 @@ impl TypeAnnotator<'_> {
             ))
     }
 
-    pub fn annotate_call_statement(
-        &mut self,
-        operator: &AstNode,
-        parameters_stmt: Option<&AstNode>,
-        ctx: &VisitorContext,
-    ) {
-        let parameters = if let Some(parameters) = parameters_stmt {
-            self.visit_statement(ctx, parameters);
-            flatten_expression_list(parameters)
+    pub fn annotate_arguments(&mut self, operator: &AstNode, arguments_node: &AstNode, ctx: &VisitorContext) {
```

Comment:
yet another nitpick: we aren't just annotating the arguments here, but also the operator (in the tail call to `update_generic_call_statement`), so I once again feel like the original method name was more fitting